irispy/spectrograph.py

```diff
@@ -208,6 +209,363 @@ def spectral_windows(self):
+
+    meta: `dict` or header object
+        Metadata associated with the sequence.
+
+    """
+    def __init__(self, data_list, common_axis, raster_positions_per_scan,
+                 first_exposure_raster_position, meta):
+        detector_type_key = "detector type"
+        # Check that meta contains required keys.
+        required_meta_keys = [detector_type_key, "spectral window",
+                              "brightest wavelength", "min wavelength", "max wavelength"]
+        if not all([key in list(meta) for key in required_meta_keys]):
+            raise ValueError("Meta must contain following keys: {0}".format(required_meta_keys))
+        # Check that all spectrograms are from same specral window and OBS ID.
+        if len(np.unique([cube.meta["OBSID"] for cube in data_list])) != 1:
+            raise ValueError("Constituent IRISSpectrogram objects must have same "
+                             "value of 'OBSID' in its meta.")
+        if len(np.unique([cube.meta["spectral window"] for cube in data_list])) != 1:
+            raise ValueError("Constituent IRISSpectrogram objects must have same "
+                             "value of 'spectral window' in its meta.")
+        # Initialize Sequence.
+        super(IRISSpectrogramSequence, self).__init__(
+            data_list, common_axis, raster_positions_per_scan,
+            first_exposure_raster_position, meta=meta)
+
+    def __repr__(self):
+        number_of_rasters = int(
+            (self.dimensions.shape[0] + self.first_exposure_raster_position) / \
+                self.raster_positions_per_scan)
+        return """IRISSpectrogramSequence
+---------------------
+{obs_repr}
+
+Sequence period: {inst_start} -- {inst_end}
+Rasters:  {n_rasters}
+Exposures per Raster: {n_steps}
+Axis Types: {axis_types}
+Sequence Shape: {seq_shape}
+
+""".format(obs_repr=_produce_obs_repr_string(self.meta),
+           inst_start=self[0]._extra_coords["time"]["value"],
+           inst_end=self[-1]._extra_coords["time"]["value"],
+           n_rasters=number_of_rasters, n_steps=self.raster_positions_per_scan,
+           axis_types=self.dimensions.axis_types[::], seq_shape=self.dimensions.shape)
+
+    def to(self, new_unit_type, copy=False):
+        """
+        Converts data, uncertainty and unit of each spectrogram in sequence to new unit.
+
+        Parameters
+        ----------
+        new_unit_type: `str`
+           Unit type to convert data to.  Three values are accepted:
+           "DN": Relevant IRIS data number based on detector type.
+           "counts": photon counts
+           "radiance": Perorms radiometric calibration conversion.
+
+        copy: `bool`
+            If True a new instance with the converted data values is return.
+            If False, the current instance is overwritten.
+            Default=False
+
+        """
+        converted_data_list = []
+        for cube in self.data:
+            converted_data_list.append(cube.to(new_unit_type))
```

Comment:
```python
converted_data_list = list(map(cube.to, self.data))
```

Reply:
that's not quite enough
```python
from functools import partial

converted_data_list = list(map(partial(cube.to, new_unit_type), self.data))
```
Also does this really need the list cast?